Source/GTMSessionFetcher.m

```diff
@@ -250,11 +247,16 @@ + (void)load {
              name:UIApplicationDidFinishLaunchingNotification
            object:nil];
 #elif GTMSESSION_RECONNECT_BACKGROUND_SESSIONS_ON_LAUNCH && TARGET_OS_OSX
-  NSNotificationCenter *nc = [NSNotificationCenter defaultCenter];
-  [nc addObserver:self
-         selector:@selector(reconnectFetchersForBackgroundSessionsOnAppLaunch:)
-             name:NSApplicationDidFinishLaunchingNotification
-           object:nil];
+  // Weakly link the application-did-finish-launching notification name, since command-line
+  // Mac clients may not link AppKit.
+  extern NSNotificationName NSApplicationDidFinishLaunchingNotification __attribute__((weak_import));
+  if (NSApplicationDidFinishLaunchingNotification != NULL) {
+    NSNotificationCenter *nc = [NSNotificationCenter defaultCenter];
+    [nc addObserver:self
+           selector:@selector(reconnectFetchersForBackgroundSessionsOnAppLaunch:)
+               name:NSApplicationDidFinishLaunchingNotification
+             object:nil];
+  }
```

Comment:
In theory, i guess we could do the `else` here and keep the old behavior, but I guess for commandline tools it doesn't really matter.

Reply:
I believe the fail case here is while it made the import weak, it AppKit still needed to be passed for linking commandline tools, so while it made the runtime not needed, it was still build breaking in that one still had to pass AppKit to link.

Reply:
Ah, I believe you are correct. I suppose that the right approach would be to use `dlsym` to load the symbol?